ff/src/fields/models/fp/montgomery_backend.rs

```diff
@@ -90,9 +90,9 @@ pub trait MontConfig<const N: usize>: 'static + Sync + Send + Sized {
     #[inline(always)]
     fn add_assign(a: &mut Fp<MontBackend<Self, N>, N>, b: &Fp<MontBackend<Self, N>, N>) {
         // This cannot exceed the backing capacity.
-        a.0.add_with_carry(&b.0);
+        let c = a.0.add_with_carry(&b.0);
         // However, it may need to be reduced
-        a.subtract_modulus();
+        a.subtract_modulus_with_carry(c);
```

Comment:
We can add a `#[doc(hidden)]` constant of the form `MODULUS_HAS_SPARE_BIT` here, and use it to make the `subtract_modulus_with_carry` optional.